### DISCUSSION TITLE
Nested config concept

### DISCUSSION THREAD

**Discussion by adamjedlicka** [storefront-x/storefront-x#145] [RFCs]
Currently the config concept uses flat structure like this:
```
config/MAGENTO_URL.ts
config/MAGENTO_GRAPHQL_ENDPOINT.ts
config/VUE_I18N_LOCALES.ts
config/VUE_I18N_ROUTE_PATHS.ts
config/CATALOG_PAGE_SIZE.ts
```
This is mainly because of two legacy reasons:

1. In the beginning, config was loaded from `.env` files, which do not allow nesting.
2. Old SFX (Vue 2) did not support nested concepts, so flat structure was the only option.

During migration form Vue 2 to Vue 3 this structure was kept for the sake of simplicity. But now when SFX supports nested concepts, and complexity of applications grow, I propose we migrate config concept to nested structure like this:
```
config/magento/url.ts
config/magento/graphqlEndpoint.ts
config/vue-i18n/locales.ts
config/vue-i18n/routePaths.ts
config/catalog/pageSize.ts
```
This will improves clarity because now configs are separated in directories based on module from which they came from, and decreases the chance of accidental overriding (collision) of config files.

**Comment by JanVanekRelitas** [comment]
I totally agree that we should use folder structure in config concept. For me most convenient way is having name of config in caps and separated with '_', like CATALOG_PAGE_SIZE.ts. I belive we should keep some more information about the config in name, not just ID.ts so there is always context when used somewhere.
